app/src/main/java/ch/epfl/sdp/appart/UserAdsActivity.java

```diff
@@ -0,0 +1,52 @@
+package ch.epfl.sdp.appart;
+
+import androidx.appcompat.widget.Toolbar;
+import androidx.lifecycle.ViewModelProvider;
+import androidx.recyclerview.widget.RecyclerView;
+import ch.epfl.sdp.appart.database.DatabaseService;
+import ch.epfl.sdp.appart.scrolling.card.Card;
+import ch.epfl.sdp.appart.scrolling.card.CardAdapter;
+import ch.epfl.sdp.appart.ad.UserAdsViewModel;
+import dagger.hilt.android.AndroidEntryPoint;
+
+import android.os.Bundle;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+@AndroidEntryPoint
+public class UserAdsActivity extends ToolbarActivity {
+
+    @Inject
+    DatabaseService database;
+    private RecyclerView recyclerView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_user_ads);
+
+        Toolbar toolbar = findViewById(R.id.UserAds_toolbar);
+        setSupportActionBar(toolbar);
+
+        UserAdsViewModel mViewModel = new ViewModelProvider(this).get(UserAdsViewModel.class);
+
+        mViewModel.initHome();

```

Comment:
Consider making initHome return a completable future and act accordingly to its result. For example, show a toast in the failing case.